mvc/controllers/areas.rst

```diff
@@ -1,8 +1,132 @@
-.. include:: /../common/stub-topic.txt
+Areas
+=====
+By `Tom Archer`
 
-|stub-icon| Areas
-===========================================
+MVC uses a folder hierarchy pattern that defines the relationship between your controllers, views, and actions. You can see this with the default wizard-generated ASP.NET MVC app where you start with a ``Controllers`` folder containing several controllers (``Account``, ``Home``, and ``Manage``). If you open the ``HomeController.cs`` file, you'll note that the ``HomeController`` controller has three actions - ``Index``, ``About``, and ``Contact`` - that return a view object without qualifying which view to return. (A fourth action - ``Error`` - is present, but it explicitly names the location of the view to return.) So, how does MVC know which view to return when its location is not explicitly stated?
 
-.. include:: /../common/stub-notice.txt
+MVC finds the associated view for a controller action by searching in two folders: ``Views/[controller][action].cshtml`` and ``Views/Shared/[action].cshtml``. Therefore, when the ``HomeController.About`` method attempts to create a view without specifying the location of the view, the MVC routing engine will look for ``Views/Home/About.cshtml`` first. If that's not found, the routing engine will look for ``Views/Shared/About.cshtml``. If neither are found, an exception is thrown.
 
-.. _issue: https://github.com/aspnet/Docs/issues/121
+This pattern works well for many Web apps. However, for larger apps, this folder hierarchy can become difficult to manage as you have controllers in one parent folder, and views in a separate parent folder. This is where `Areas` come in. Areas enable you to partition your project such that controllers and their views can be stored under a parent folder that defines their logical grouping.
```

Comment:
> as you have controllers in one parent folder, and views in a separate parent folder.

This isn't really the reason things get difficult to manage. The reason is that as apps get bigger you end up with a lot of controllers and views, which makes the app difficult to reason about. Areas let you partition your app into logic chunks that are smaller and easier to manage. It's this logical partitioning that is important, not the folder structure (note that the folder structure really only matters for views in MVC 6).